src/physics/physics_model.cpp

```diff
@@ -143,28 +185,39 @@ ::convective_eigenvalues (
     const std::array<real,nstate> &conservative_soln,
     const dealii::Tensor<1,dim,real> &normal) const
 {
-    // TO DO: Update for when nstate > nstate_baseline_physics
     std::array<real,nstate> eig;
-    if(nstate==nstate_baseline_physics) {
+    if constexpr(nstate==nstate_baseline_physics) {
         eig = physics_baseline->convective_eigenvalues(conservative_soln, normal);
     } else {
-        // TO DO, make use of the physics_model object for nstate>nstate_baseline_physics
-        std::abort();
+        eig = model->convective_eigenvalues(conservative_soln, normal);
+        std::array<real,nstate_baseline_physics> baseline_conservative_soln;
+        for(int s=0; s<nstate_baseline_physics; ++s){
+            baseline_conservative_soln[s] = conservative_soln[s];
+        }
+        std::array<real,nstate_baseline_physics> baseline_eig = physics_baseline->convective_eigenvalues(baseline_conservative_soln, normal);
+        for(int s=0; s<nstate_baseline_physics; ++s){
+            eig[s] += baseline_eig[s];
+        }  
```

Comment:
As discussed with Siva, for now, I just leave a note mentioning this part only supports the zero additional convective flux in the baseline physics.
(in this case the baseline eigenvalues in model are zeros)

Reply:
You could add an something like `if (eig[s]!=0.0){ pcout << "Error: PhysicsModel does not currently support additional convective flux terms. Aborting..." << std::endl; std::abort();}` before you do `eig[s] += baseline_eig[s];`. This way the problem will arise if someone does that

Reply:
fixed as suggested, now it will check if the entries of baseline_physics is zero, the error will show and the code will be aborted if none zero entries are given.